qos/evm/context.go

```diff
@@ -195,3 +197,10 @@ func preSelectedEndpoint(
 
 	return protocol.EndpointAddr(""), fmt.Errorf("singleEndpointSelector: endpoint %s not found in available endpoints", preSelectedEndpointAddr)
 }
+
+// setPreSelectedEndpointAddr assigns the endpoint address to be used for hydrator checks.
+// It is called to override the endpoint selection process with a specific endpoint.
+// and is used to enforce performing quality checks on a specific endpoint.
+func (rc *requestContext) setPreSelectedEndpointAddr(endpointAddr protocol.EndpointAddr) {
```

Comment:
Can you add a TODO_IMPROVE to have a "golden fallback endpoint" that's configurable via env variables to PATH?